samples/bookinfo/README.md

```diff
@@ -1,33 +1,98 @@
+Step 1/16 : FROM python:3.7.7-slim
+3.7.7-slim: Pulling from library/python
+8559a31e96f4: Pull complete
+...
+Successfully built 1b293582cc2e
+Successfully tagged shamsher31/examples-bookinfo-ratings-v2:1.16.3
+Successfully tagged shamsher31/examples-bookinfo-ratings-v2:latest
+```
 
 The bookinfo versions are different from Istio versions since the sample should work with any version of Istio.
 
-## Update docker images in the yaml files
+## Push docker images to docker hub
+
+After the local build is successful, you need to update the YAML file with the latest tag that you used during the build eg: `1.16.3`.
+
+Run the following script to build the docker images, push them to docker hub, and to update the YAML files in one step.
 
 ```bash
-sed -i "s/\(istio\/examples-bookinfo-.*\):[[:digit:]]\.[[:digit:]]\.[[:digit:]]/<your docker image with tag>/g" */bookinfo*.yaml
+./build_push_update_images.sh <version> <prefix>
 ```
 
-## Push docker images to docker hub
-
-One script to build the docker images, push them to docker hub and to update the yaml files
+For example:
 
 ```bash
-build_push_update_images.sh <version>
+$ ./build_push_update_images.sh 1.16.3 --prefix=shamsher31
+...
+1.16.3: digest: sha256:70634d3847a190b9826975c8 size: 3883
+Pushing: shamsher31/examples-bookinfo-reviews-v2:1.16.3
+The push refers to a repository [docker.io/shamsher31/examples-bookinfo-reviews-v2]
+...
 ```
 
+Verify that expected tag eg: `1.16.3` is updated in `platform/kube/bookinfo*.yaml` files.
+
 ## Tests
 
+Test that the bookinfo samples work with the latest tag eg: `1.16.3` that you pushed.
+
+```bash
+$ cd ../../
+$ kubectl apply -f samples/bookinfo/platform/kube/bookinfo.yaml
+serviceaccount/bookinfo-details created
+deployment.apps/details-v1 created
+serviceaccount/bookinfo-ratings created
+...
+```
+
+Wait for all the pods to be in `Running` start.
+
+```bash
+$ kubectl get pods
+NAME                              READY   STATUS    RESTARTS   AGE
+details-v1-7f556f5c6b-485l2       2/2     Running   0          10m
+productpage-v1-84c8f95c8d-tlml2   2/2     Running   0          10m
+ratings-v1-66777f856b-2ls78       2/2     Running   0          10m
+reviews-v1-64c47f4f44-rx642       2/2     Running   0          10m
+reviews-v2-66b6b95f44-s5nt6       2/2     Running   0          10m
+reviews-v3-7f69dd7fd4-zjvc8       2/2     Running   0          10m
+```
+
+Once all the pods are in the `Running` state. Test if the bookinfo works through cli.
+
+```bash
+$ kubectl exec -it "$(kubectl get pod -l app=ratings -o jsonpath='{.items[0].metadata.name}')" -c ratings -- curl productpage:9080/productpage | grep -o "<title>.*</title>"
+<title>Simple Bookstore App</title>
+```
+
+You can also test it by hitting producpage in the browser.
+
+```bash
+http://192.168.39.116:31395/productpage
+```
+
+You should see the following in the browser.
+
+![star](https://user-images.githubusercontent.com/2920003/86032538-212ff900-ba55-11ea-9492-d4bc90656a02.png)
+
```

Comment:
I think this sounds good. Maybe add a step to note that this testing was completed successfully in the PR with the changes so reviewers know it was done.

Now we are at a point where we can go one of two ways:

- Merge the PR. Then build the official images and change the version in the yaml in a different PR.
- Request a new official set of images be built and pushed, and add another commit to the original PR with the version changes.

I'm in favor of the second since it keeps everything in one PR. @frankbu, what do you think?

Reply:
I agree with @ericvn, the second option is best. Push the official image before merging the PR.

Reply:
@ericvn does this sounds good add at the end of these guidelines?

> Note: If everything works as mentioned above, Request a new official set of images be built and pushed from the reviewer, and add another commit to the original PR with the version changes.